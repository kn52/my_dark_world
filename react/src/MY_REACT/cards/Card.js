import React from 'react';
import Home from '../../Home/Home';
import CheckCardType from './CheckCardType';

class Card extends React.Component {

    constructor(props){
        super(props);
        this.state ={
            cardType:"",
            cardNumber:""
        }
    }

    handleCardType = async (e) => {
        var number= "";
        var cType=[]
        number=e.target.value;
        if(number.length >=6) {
            cType=CheckCardType.checkCardType(number.trim());
            await this.setState({cardType:cType});
        }else{ 
            await this.setState({cardType:""}); 
        }
    }
    
    componentDidMount(){
        // localStorage.getItem("password") === null && window.location.replace("/"); 
    }

    render() {
        return (
            <>
                <div><Home /></div>
                <div style={{width:'100%',display:'flex',justifyContent:'center',flexDirection:'column', alignItems:'center', color: "black", height: "80%", background: "white" }}>
                    <input data-testid="input" type="text" value={this.state.cardNumber} maxLength="8" onChange={(e) => {
                        this.setState({ cardNumber: e.target.value }); this.handleCardType(e)
                    }} />
                    <br />
                    <label>Card Number: {this.state.cardNumber}</label>
                    <br />
                    <label>Card type</label>
                    <br />
                    {this.state.cardType.length !== 0 && <img src={this.state.cardType} width="70px" height="60px" alt="" />}
                </div>
            </>
        )
    }
}

export default Card;