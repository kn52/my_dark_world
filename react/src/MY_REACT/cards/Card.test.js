import { shallow, configure } from "enzyme";
import Adapter from 'enzyme-adapter-react-16';
import toJson from "enzyme-to-json";
import React from "react";
import { render, unmountComponentAtNode } from "react-dom";
import { act } from "react-dom/test-utils";

import Card from './Card';


configure({ adapter: new Adapter() });
let container = null;
beforeEach(() => {
  container = document.createElement("div");
  document.body.appendChild(container);
});

afterEach(() => {
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

test('renders card component', () => {
  act(()=>{
    render(<Card />,container);
  })
});

describe("Card Component", () => {
  it("should render my component", () => {
    const wrapper = shallow(<Card />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
  
describe("card Component props", () => {
    const wrapper = shallow(<Card />);
    const form = wrapper.find('input');
    const formProps = form.props();
    
    it("should have a <input> with length", () => {
      expect(form).toHaveLength(1);
    });
    
    it("should have a <input> with properly type prop", () => {
      expect(formProps.type).toEqual("text");
    });

    it("should have a <div> with properly onChange function prop", () => {
      expect(typeof formProps.onChange).toBe("function");
    });
});

describe("Calculator tests", () => {
  
  // function filterByTerm(inputArr, searchTerm) {
  //   return inputArr.filter(function(arrayElement) {
  //   return arrayElement.url.match(searchTerm);
  //   });
  // }
    
  const mathOperations = {
    sum: function (a, b) {
      return a + b;
    },
  
    diff: function (a, b) {
      return a - b;
    },
    product: function (a, b) {
      return a * b
    }
  }
  
  it('adding 1 + 2 should return 3', () => {
    expect(mathOperations.sum(1, 2)).toBe(3);
  });
})