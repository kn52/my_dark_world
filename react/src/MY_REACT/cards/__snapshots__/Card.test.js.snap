// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Component should render my component 1`] = `
<div
  style={
    Object {
      "background": "white",
      "color": "black",
      "height": "80%",
      "left": "0px",
      "width": "50%",
    }
  }
>
  <input
    data-testid="input"
    maxLength="8"
    onChange={[Function]}
    type="text"
    value=""
  />
  <br />
  <label>
    Card Number: 
  </label>
  <br />
  <label>
    Card type
  </label>
  <br />
</div>
`;
